s3tests/functional/test_s3.py

```diff
@@ -8266,3 +8266,114 @@ def test_sse_kms_barb_transfer_13b():
     if 'kms_keyid' not in config['main']:
         raise SkipTest
     _test_sse_kms_customer_write(13, key_id = config['main']['kms_keyid'])
+
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='Test Bucket Policy')
+@attr(assertion='succeeds')
+@attr('bucket-policy')
+def test_bucket_policy():
+    bucket = get_new_bucket(name='my-bucket')
+    key = bucket.new_key('asdf')
+    key.set_contents_from_string('asdf')
+    l = bucket.list()
+    policy_document =	\
+    "{                             \
+        \"Version\": \"2012-10-17\",        \
+        \"Statement\": [{                   \
+        \"Effect\": \"Allow\",            \
+        \"Principal\": {\"AWS\": \"*\"},  \
+        \"Action\": \"s3:ListBucket\",    \
+        \"Resource\": [                   \
+            \"arn:aws:s3:::my-bucket\",  \
+            \"arn:aws:s3:::my-bucket/*\" \
+          ]                                 \
+        }]                                  \
+     }"
+    bucket.set_policy(policy_document)
+
+    new_conn = boto.s3.connection.S3Connection(
+        aws_access_key_id=s3['alt'].aws_access_key_id,
+        aws_secret_access_key=s3['alt'].aws_secret_access_key,
+        is_secure=s3['alt'].is_secure,
+        port=s3['alt'].port,
+        host=s3['alt'].host,
+        calling_format=s3['alt'].calling_format,
+        )
+    b = new_conn.get_bucket('my-bucket')
+    b.get_all_keys()
+
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='Test Bucket Policy and ACL')
+@attr(assertion='fails')
+@attr('bucket-policy')
+def test_bucket_policy_acl():
+    bucket = get_new_bucket(name='my-bucket-2')
+    key = bucket.new_key('asdf')
+    key.set_contents_from_string('asdf')
+    l = bucket.list()
+    policy_document =   \
+    "{                             \
+        \"Version\": \"2012-10-17\",        \
+        \"Statement\": [{                   \
+        \"Effect\": \"Deny\",            \
+        \"Principal\": {\"AWS\": \"*\"},  \
+        \"Action\": \"s3:ListBucket\",    \
+        \"Resource\": [                   \
+            \"arn:aws:s3:::my-bucket-2\",  \
+            \"arn:aws:s3:::my-bucket-2/*\" \
+          ]                                 \
+        }]                                  \
+     }"
+    bucket.set_canned_acl('authenticated-read')
+    bucket.set_policy(policy_document)
+
+    new_conn = boto.s3.connection.S3Connection(
+        aws_access_key_id=s3['alt'].aws_access_key_id,
+        aws_secret_access_key=s3['alt'].aws_secret_access_key,
+        is_secure=s3['alt'].is_secure,
+        port=s3['alt'].port,
+        host=s3['alt'].host,
+        calling_format=s3['alt'].calling_format,
+        )
+    e = assert_raises(boto.exception.S3ResponseError, new_conn.get_bucket, 'my-bucket-2')
+    eq(e.status, 403)
+    eq(e.reason, 'Forbidden')
+    eq(e.error_code, 'AccessDenied')
+
+@attr(resource='bucket')
+@attr(method='get')
```

Comment:
Actually, the intention is to check the side-effect of setting a policy on the Bucket List operation, hence I think 'get' will be appropriate to describe this.